installer/DetailsHandler.php

```diff
@@ -327,6 +333,59 @@ function isFillInConfig($SessionStatus, $inputType)
         return trim($returnValue);
     }
 
+    /**
+     * Check whether registration data are valid
+     * @param $SessionStatus
+     * @param $inputType
+     * @param bool $invalidInput
+     * @return string
+     */
+    function isFillInConfigRegister($SessionStatus, $inputType, $invalidInput = false)
+    {
+        if ($inputType == 'Organization Email Address ') {
+            $invalidMessage = "Organization Email Address invalid.";
+        }
+        if (!isset($SessionStatus) || trim($SessionStatus) === '') {
+
+            if (!$invalidInput) {
+                $returnValue = $this->takeUserInput("Please enter $inputType : ");
+            } else {
+                $returnValue = $this->takeUserInput($invalidMessage . "Please enter $inputType : ");
+            }
+
+            return $returnValue ? ($this->isValidUserInput($inputType, $returnValue) ? $returnValue : $this->isFillInConfigRegister(null, $inputType, true)) : ($this->isFillInConfigRegister(null, $inputType));
+
+        } else {
+            $returnValue = $SessionStatus;
+        }
+
+        return $returnValue;
+    }
+
+    /**
+     * Validate input
+     * @param $inputType
+     * @param $inputValue
+     * @return bool|mixed
+     */
+    function isValidUserInput($inputType, $inputValue) {
+        if ($inputType == 'Organization Email Address ') {
+            return filter_var($inputValue, FILTER_VALIDATE_EMAIL);
+        }
+
+        if ($inputType == 'contact number') {
+            if (!($inputValue == '')) {
+                if(preg_match("/^\+?[0-9]+$/", $inputValue)) {
+                    $_SESSION['defUser']['contactNumber'] = $inputValue;
+                    return true;
+                }
+                return false;
```

Comment:
populate error messages and don't return it .. $this->error['contact-number'] = true

Reply:
updated